build_scripts/desktop/finish_darwin.sh

```diff
@@ -0,0 +1,107 @@
+#!/bin/bash
+#
+# Script to finish packaging individual iOS/Mac libraries into
+# universal libraries, and then into frameworks.
+set -e
+
+if [[ "$(uname)" != "Darwin" ]]; then
+    echo "Error: Frameworks and universal libraries can only be built on a Mac host."
+    exit 2
+fi
+
+packagepath=$1
+includepath=$2
+binutils=$3
+os=$4
+
+if [[ -z "${os}" ]]; then
+    os=darwin
+fi
+
+if [[ -z "${packagepath}" || -z "${includepath}" || -z "${binutils}" ]]; then
+    echo "Usage: $0 <packaged iOS/Mac SDK path> <include file path> <binutils path> [darwin|ios]"
+    exit 2
+fi
+
+if [[ ! -d "${packagepath}/libs/${os}" ]]; then
+    echo "Can't find packaged iOS/Mac SDK in ${packagedpath}/libs/darwin"
+fi
+
+if [[ ! -r "${includepath}/firebase/app.h" ]]; then
+    echo "Can't find include files in ${includepath}"
+    exit 2
+fi
+
+if [[ ! -x "${binutils}/ar" ]]; then
+    echo "Packaging tools not found at path '${binutils}'."
+    exit 2
+fi
+
+origpath=$(pwd -P)
+packagepath=$(cd ${packagepath}; pwd -P)
+includepath=$(cd ${includepath}; pwd -P)
+binutils=$(cd ${binutils}; pwd -P)
+
+declare -a architectures
+for a in $(cd "${packagepath}"/libs/"${os}"; find . -depth 1 -type d | sed 's|^\./||'); do
+    if [[ $a != "universal" ]]; then
+	architectures+=($a)
+    fi
+done
+
+if [[ ! -d "${packagepath}/libs/${os}/universal" ]]; then
+    echo "Repackaging libraries using Mac libtool..."
+    # Don't lipo if we already have. TODO: remove this, debugging only.
```

Comment:
Just incase you forgot to remove this bit as the comment mentions its for debugging only. Comment "Dont lipo if we already have." seems incomplete.

Reply:
Reworded and moved to the correct location. Basically it's a case that can only come up when debugging but I made that clearer in the comment.